resteasy-client/src/main/java/org/jboss/resteasy/client/jaxrs/internal/proxy/ClientInvoker.java

```diff
@@ -144,13 +144,16 @@ protected Object invokeAsync(final Object[] args)
 
    protected Object invokeSync(Object[] args)
    {
-      ClientInvocation request = createRequest(args);
-      ClientResponse response = (ClientResponse)request.invoke();
+      ClientInvocationBuilder requestBuilder = createRequest(args);
+      ClientInvocation request = (ClientInvocation)requestBuilder.build(httpMethod);
+      request.setClientInvoker(this);
+
+      ClientResponse response = request.invoke();
       ClientContext context = new ClientContext(request, response, entityExtractorFactory);
       return extractor.extractEntity(context);
    }
 
-   protected ClientInvocation createRequest(Object[] args)
+   protected ClientInvocationBuilder createRequest(Object[] args)
```

Comment:
We need to be careful about backward compatibility in 3.6 PR. This changes protected method signature from public class from public WF module.

@asoldano + @ronsigal WDYT about this?

Reply:
Good point, @marekkopecky. It's probably not a problem, but it's good to be safe. @fabienrousseau, can you just move the line
```
builder.setInvocation(request);
```
to the end of createRequest(), where it would become
```
builder.setInvocation(clientInvocation);
```
?